### PR TITLE
change api key config

### LINKED ISSUE #1104: API Key got from API Manager fail in microgateway
### Description:

Since the apikey in API Manager doesnot contain the audience claim, we fails to validate that apikey.
 
### Steps to reproduce:
get apikey from api manager and use it as apikey in mgw

### Affected Product Version:

api manager 3.1.0
mgw 3.1.0-beta

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### LINKED ISSUE #1105: Change the api key issuers api list configuration format
### Describe your problem(s)

Current configuration needs to index apis with 1,2,.. as below.
```
[apikey.issuer.apis.1]
      name="name"
      versions="1.0.0, v2"
[apikey.issuer.apis.2]
      name="name"
      versions="1.0.0, v2"
```

### Describe your solution

```
[[apikey.issuer.api]]
      name="name"
      versions="1.0.0, v2"
[[apikey.issuer.api]]
      name="name2"
      versions="1.0.0"
```

Change it to the following;

### How will you implement it

---
#### Suggested Labels:

3.1.0

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Purpose

Removing aud claim from api key issuer and validator.
Change api key apis configuration format to below;
```
[[apikey.issuer.api]]
  name="name"
  versions="1.0.0, v2"
[[apikey.issuer.api]]
  name="name2"
  versions="1.0.0"
```

### Issues

Fixes #1105
Fixes #1104 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal [hunk 1]
```diff
@@ -760,7 +760,6 @@ public function initAuthHandlers() {
     //Initializes apikey handler
     jwt:JwtValidatorConfig apiKeyValidatorConfig = {
         issuer: getConfigValue(API_KEY_INSTANCE_ID, ISSUER, DEFAULT_API_KEY_ISSUER),
-        audience: getConfigValue(API_KEY_INSTANCE_ID, AUDIENCE, DEFAULT_AUDIENCE),
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal]: So the fix for #1104 if removing audience validation for any apiKey (APIM or MGW Key)? 
Anyway I'd be extremely happy if you sent these two fixes in two different commits.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/api_key_utils.bal [hunk 2]
```diff
@@ -187,3 +167,27 @@ public function getAPIKeyAuth() returns [string, string] | error {
...
+
+public function getAPIsforAPIKey() returns json[] {
+    json[] apis = [];
+    map<anydata>[] | error apiList = map<anydata>[].constructFrom(config:getAsArray(API_KEY_ISSUER_API));
+    if (apiList is map<anydata>[] && apiList.length() > 0) {
+        foreach map<anydata> apiData in apiList {
+            anydata apiName = apiData[NAME];
+            anydata apiVersions = apiData[VERSIONS];
+            if (apiName is string) {
+                if (!(apiVersions is string) || (stringutils:equalsIgnoreCase("*", apiVersions))) {
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/utils/api_key_utils.bal]: Shall we use simple equals here? No need of `equalsIgnoreCase` when comparing a special `char`?
```suggestion
                if (!(apiVersions is string) || "*" == apiVersions) {
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal (hunk 1 above)
Comment (AmaliMatharaarachchi) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal]: Yes, since we removed aud from validation, no point in issuing an API Key with aud claim. Added as 2 commits as you requested.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/api_key_utils.bal (hunk 2 above)
Comment (AmaliMatharaarachchi) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/utils/api_key_utils.bal]: done
